### PR TITLE
Update velox config and operator docs

### PR DESCRIPTION
**Pull request by xiaoxmeng**
- Update operator doc for PartitionedOutputNode
- Update config doc for spilling configs
- Remove some legacy code

### CODE REVIEW

File: velox/docs/configs.rst [hunk 1]
```diff
@@ -210,11 +210,17 @@ Spilling
      - integer
      - 29
      - The start partition bit which is used with `spiller_partition_bits` together to calculate the spilling partition number.
-   * - spiller_partition_bits
+   * - join_spiller_partition_bits
      - integer
      - 2
-     - The number of bits used to calculate the spilling partition number. The number of spilling partitions will be power of
-       two. At the moment the maximum value is 3, meaning we only support up to 8-way spill partitioning.
+     - The number of bits used to calculate the spilling partition number for hash join. The number of spilling partitions will
```
Comment (mbasmanova) [inline velox/docs/configs.rst:216]: It is a bit misleading.

First it says, "The number will be power of two", but then it is "value is 3".

File: velox/docs/develop/operators.rst [hunk 2]
```diff
@@ -401,12 +401,12 @@ distribution fields.
 
    * - Property
      - Description
+   * - kind
+     - Specifies output buffer types: kPartitioned, kBroadcast and kArbitrary. For kPartitioned type, rows are partitioned and each sent to corresponding destination partition; For kBroadcast type, rows are not partitioned and sent to all the destination partitions; For kArbitrary type, rows are not partitioned and each sent to any one of the destination partitions.
```
Comment (mbasmanova) [inline velox/docs/develop/operators.rst:405]: perhaps, replace ; with .

**Review by mbasmanova — APPROVED** [review]
Thank you for keeping the documentation up to date.

File: velox/docs/configs.rst (hunk 1 above)
Comment (xiaoxmeng) [reply velox/docs/configs.rst]: Yeah, I updated to "The number of bits (N) used to calculate the spilling partition number for hash join: 2 ^ N."

**Pull request closed without merging**